### PR TITLE
[Community, WIP] A pipeline to run Stable Diffusion with k-diffusion by crowsonkb

### PR DESCRIPTION
**Pull request by BurguerJohn**
The code run from start to finish, but it's giving pretty bad outputs.
I did try to undestand both codes and I'm currently stuck in what could be possible wrong.
Perhaps someone can shed some light of what is wrong with my code?

### CODE REVIEW

File: examples/community/k_stable_diffusion.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+from diffusers.configuration_utils import FrozenDict
+from diffusers import AutoencoderKL, DiffusionPipeline, DDIMScheduler, LMSDiscreteScheduler, PNDMScheduler, UNet2DConditionModel
+from diffusers.pipelines.stable_diffusion import StableDiffusionPipelineOutput
+from diffusers.utils import deprecate, logging
+from diffusers.pipelines.stable_diffusion.safety_checker import StableDiffusionSafetyChecker
+
+import k_diffusion as K
+import torch.nn as nn
+
+class KWrapper(nn.Module):
```
Comment (patrickvonplaten) [inline examples/community/k_stable_diffusion.py:17]: Maybe we could also add this here: https://github.com/crowsonkb/k-diffusion/blob/f4e99857772fc3a126ba886aadf795a332774878/k_diffusion/external.py#L131 -> looks nice!

**Comment by patrickvonplaten** [comment]
This looks very nice to me already! Think we can merge this whenever you'd like. Two things maybe:
- Could you run `make style` to fix the `check_code_quality` test?
- Could you maybe add a short code example of how to run your community pipeline here: https://github.com/huggingface/diffusers/tree/main/examples/community#example-usages

**Comment by patrickvonplaten** [comment]
Very excited about this PR - think it can generate quite some usage :-)

**Comment by BurguerJohn** [comment]
> 



> This looks very nice to me already! Think we can merge this whenever you'd like. Two things maybe:
> 
> * Could you run `make style` to fix the `check_code_quality` test?
> * Could you maybe add a short code example of how to run your community pipeline here: https://github.com/huggingface/diffusers/tree/main/examples/community#example-usages

Sure, can do both once I get some time today. But the main problem is that is just output garbage for now 😅

**Comment by patrickvonplaten** [comment]
Sounds good! Yeah the schedulers can be complicated sometimes! `ipdb` and `print(x.abs().sum())` are usually my best friends when debugging them step-by-step

**Comment by BurguerJohn** [comment]
@patrickvonplaten Gonna have to give it up on this for now, everything look good with the code, looking at the examples, it should be working, but it's not =/
Once I get some free time again, may give it another shot on it.
I did run the make style script on the last push, but it seen that it still failed the code quality script, maybe is something wrong in my end.

**Comment by patrickvonplaten** [comment]
Cool, let's see whether anybody in the community is interested in picking it up - otherwise this might also be interesting for @anton-l

**Comment by patrickvonplaten** [comment]
Not sure if anybody has time for this... but, we could also definitely host a StableDiffusionKDiffusionPipeline in `src/diffusers/pipelines` that depends of k-diffusion -> could be nice to see if we can nicely merge diffusers and k-diffusion this way! 

Would be very open for such a pipeline :-) In short it could be as simple as just storing the model components in the pipeline (no "scheduler" component) and then let k-diffusion do the whole denoising process. Very happy to help if anybody is interested!
